### DISCUSSION TITLE
CProcessModel tutorial

### DISCUSSION THREAD

**Discussion by FilipDepta** [lava-nc/lava#485] [Q&A] [answered]
Hi all, as this is my first discussion on Lava!
I can't find any tutorial about writing a custom CProcessModel. There is this note in the very first tutorial that CProcessModel will be discussed later. However, I couldn't find it in the tutorials and in the docs on the lava-nc web. Have I missed something? Are there any docs that explain sth about CProcssModels?

Thanks in advance!

**Comment by weidel-p** [comment, marked as answer]
Hi @FilipDepta,

do you have access to the lava-loihi extension (or in other words, are you engaged INRC member)? 

If so, you will find a tutorial called "tutorial00_tour_through_lava" in the `tutorials/end_to_end` folder of the lava-loihi extension which shows the usage of a CProcModel. Alternatively, there are examples for CProcModels in the process library of the lava-loihi extension. Maybe take a look at `embedded_io/abstract_adapter.py`.

**Reply by FilipDepta** [reply]
Thank you for the answer.
Currently, I'm only an affiliate, not an engaged INRC member.

I have written my own, simple, spiking framework and I want to rewrite this to C, but using Lava processes as a convenient environment for such a framework (maybe using some other accelerators). Do you think this is a good idea for utilizing the Lava or maybe better stick to my own framework structure?

**Reply by weidel-p** [reply]
This is hard to answer, as I do not know what you are planning to use it for. In general, I do understand that it seems attractive to develop your own SNN framework, but as someone who has been involved in the development of several SNN frameworks (NEST, Rockpool, Sinabs, and now Lava), I have to advocate the benefits of using an established, open-source tool. These tools are well tested, easy to use and efficient. 

If you want to use lava and get access to the lava-loihi extension and also to neuromorphic hardware, you can apply for access here: 
https://intel-ncl.atlassian.net/wiki/spaces/INRC/pages/1784807425/Join+the+INRC

**Reply by FilipDepta** [reply]
Well, I started with Nengo, but then I've written my own framework for elasticity - it's the easiest (therefore the fastest) way to make changes in code while working on my own learning method, despite it isn't as efficient as bigger frameworks. Then Lava emerged and I wanted to switch to it, but I would like to write quite fast implementations for CPU (on the LoihiSim the monitoring slows down the simulation dramatically).

I consider joining INRC as an engaged member, but currently, my idea isn't clear enough to submit the application, I think.
Anyway, thank you for the discussion and advice, maybe I'm wrong, and should apply right now :)
